### PR TITLE
Add Db#watch definition (mongodb)

### PR DESCRIPTION
**Pull request by dvvc**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: http://mongodb.github.io/node-mongodb-native/3.1/api/Db.html#watch

### CODE REVIEW

**Comment by HosseinAgha** [comment]
Hi @dvvc  
We are currently supporting node driver v3.3. Could you please refer to [v3.3 driver docs](http://mongodb.github.io/node-mongodb-native/3.3/api/Db.html#watch) and check if you need to make any updates?

**Review by CaselIT — APPROVED** [review]

**Comment by HosseinAgha** [comment]
Hi @CaselIT.  
I forgot to submit my pending reviews 2 days ago. I think the PR does not follow docs (specially `startAfter` option).

File: types/mongodb/index.d.ts
Comment (CaselIT) [reply types/mongodb/index.d.ts]: I missed this. I agree

**Review by CaselIT — CHANGES_REQUESTED** [review]
see comments of @HosseinAgha

**Comment by CaselIT** [comment]
There seem to be no way to un-approve a pr.

**Comment by dvvc** [comment]
Thanks @HosseinAgha and @CaselIT, will review and update the PR

**Comment by dvvc** [comment]
Regarding the change in parameter types, which would affect Collection#watch and MongoClient#watch, should that be part of this same PR, or should I address that in its own PR?

**Comment by CaselIT** [comment]
I think this one is fine 👍

**Comment by CaselIT** [comment]
mmm, I think the bot has mistaken the emoji as approval. please wait before merging I think @dvvc is making some changes

**Comment by amcasey** [comment]
Adding tag so we don't get confused by the bot's error.

**Review by mrcrane — CHANGES_REQUESTED** [review]
Marking as changes requested until the comments above are addressed

**Pull request closed without merging**
